app/controllers/agent/ManageAgentController.scala

```diff
@@ -142,59 +158,128 @@ class ManageAgentController @Inject()(
                                        )
       ipPropertyLinksCount = propertyLinks.total
     } yield {
-      (ipPropertyLinksCount, agentToBeManagedOpt) match {
-        case (0, Some(agent)) if agent.propertyCount == 0 =>
-          //IP has no property links but still has an agent
-          Some(
-            removeAgentFromOrganisationView(
-              submitAgentAppointmentRequest,
-              agent.representativeCode,
-              agent.name,
-              calculateBackLink(organisationsAgents, agent.representativeCode)))
-        case (1, Some(agent)) if agent.propertyCount == 0 =>
-          //IP has one property link but agent is not assigned
-          Some(
-            manageAgentView(
-              submitManageAgentForm,
-              ManageAgentOptions.onePropertyLinkNoAssignedAgentsOptions,
-              agent,
-              calculateBackLink(organisationsAgents, agent.representativeCode)))
-        case (1, Some(agent)) if agent.propertyCount == 1 =>
-          //IP has one property link and agent is assigned to that property
-          Some(
-            unassignAgentFromPropertyView(
-              submitAgentAppointmentRequest,
-              agent,
-              calculateBackLink(organisationsAgents, agent.representativeCode)))
+      agentToBeManagedOpt match {
+        case None => throw new NotFoundException("Agent is not available")
```

Comment:
Does this need to throw an exception here, or can it return None, as the method returns an optional view and the NotFound is handled above?